docs/firefly-api-overview.md

```diff
@@ -0,0 +1,551 @@
+
+- `firefly.util.image.CCUtil.getImageCoords(plot,pt` - get image coordinates of the point
+- `firefly.util.image.CCUtil.getWorldCoords(plot,pt)` - get world coordinates of the point
+- `firefly.util.image.CCUtil.getViewPortCoords(plot,pt)` - get viewport coordinates of the point
+- `firefly.util.image.CCUtil.getScreenCoords(plot,pt)` - get screen coordinates of the point
+
+| Parameter  | Type | Description |
+| ---------- | ---- | ----------- |
+| plot | Object | Firefly plot object |
+| pt  | Object | Firefly point object |
+
+*Example:*
+```js
+function getImagePt(pt) {
+    const activePlot = firefly.util.image.getPrimePlot();
+    return pt ? firefly.util.image.CCUtil.getImageCoords(activePlot, pt) : undefined;
+}
+```
+
+#####**firefly.util.image.serializeSimpleRangeValues** method
+
+`firefly.util.image.serializeSimpleRangeValues(stretchType,lowerValue,upperValue,algorithm)` - serialize a stretch request into a string, for use with the "RangeValues" parameter
+
+| Parameter  | Type | Description |
+| ---------- | ---- | ----------- |
+| stretchType | string | the type of stretch may be 'Percent', 'Absolute', 'Sigma' |
+| lowerValue  |number | lower value of stretch, based on stretchType |
+| upperValue  |number | upper value of stretch, based on stretchType |
+| algorithm | string | algorithm the stretch algorithm to use, may be 'Linear', 'Log', 'LogLog', 'Equal', 'Squared', 'Sqrt' |
+| *return* | string | a serialized version of range values to be passed as a viewer parameter |
+
+*Example:*                               
+```js
+firefly.getViewer().showImage('imageDiv', {
+    plotId: 'p1',
+    Service: 'TWOMASS',
+    Title  : '2mass from service',
+    SurveyKey  : 'k',
+    WorldPt    : '10.68479;41.26906;EQ_J2000',
+    RangeValues : firefly.util.image.serializeSimpleRangeValues("Sigma",-1,2,"Linear"),
+    SizeInDeg  : '.12'
+});
+```
+
+`firefly.util.image.initAutoReadout(readoutComponent, props)` - initialize the auto readout.
+
+To use minimal readout, do the following:
+
+*Example:*                               
+```js
+    firefly.util.image.initAutoReadout(ui.DefaultApiReadout,
+        {
+            MouseReadoutComponent:firefly.ui.PopupMouseReadoutMinimal, 
+            showThumb:false,
+            showMag:false
+        }
+    );
+```
+
+###Region Support
+
+#####**firefly.action.dispatchCreateRegionLayer** method
+
+`firefly.action.dispatchCreateRegionLayer(regionId, layerTitle, fileOnServer ='', regionAry=[], plotId = [])` - overlay region data on an image plots with the given ids
+
+| Parameter  | Type | Description |
+| ---------- | ---- | ----------- |
+| regionId | string | region layer id |
+| layerTitle | string | title for the layer, displaying regions |
+| fileOnServer | string | region file on the server |
+| regionAry | array | an array of strings, each describing a ds9 region |
+| regionLayerId | string | region layer id |
+| plotId | string or array | a plot id or an array plot ids |
```

Comment:
for dispatchCreateRegionLayer, if no plotId is given, then the region layers is created on all plots. Should this optional setting be described?